lib/Feature/XmlParser.cpp

```diff
@@ -62,24 +77,26 @@ void XmlParser::parseConfigurationOption(xmlNode *N, bool Num = false) {
           Name, std::make_unique<NumericFeature>(
                     Name, Opt,
                     std::variant<std::pair<int, int>, std::vector<int>>(
-                        std::make_pair(MinValue, MaxValue))));
+                        std::make_pair(MinValue, MaxValue)),
+                    std::move(Loc)));
 
     } else {
       Features.try_emplace(
           Name, std::make_unique<NumericFeature>(
                     Name, Opt,
-                    std::variant<std::pair<int, int>, std::vector<int>>(Vals)));
+                    std::variant<std::pair<int, int>, std::vector<int>>(Vals),
+                    std::move(Loc)));
     }
   } else {
-    Features.try_emplace(Name, std::make_unique<BinaryFeature>(Name, Opt));
+    Features.try_emplace(
```

Comment:
The try_emplace could 'fail' and not insert the new element if the key is already present. Maybe we should check the return value with an assert.
```
auto emplaced = Features.try_emplace(...)
assert(emplaced.second && "Feature value could not be inserted, values was already present.");
```